course03/project01/balint/application.cc

```diff
@@ -0,0 +1,257 @@
+#include "cinder/app/App.h"
+#include "cinder/app/RendererGl.h"
+#include "cinder/gl/gl.h"
+
+using namespace ci;
+using namespace ci::app;
+
+// We'll create a new Cinder Application by deriving from the App class.
+class BasicApp : public App
+{
+ public:
+  // Cinder will call 'keyDown' when the user presses a key on the keyboard.
+  // See also: keyUp.
+  void keyDown(KeyEvent event) override;
+
+  // Cinder will call 'draw' each time the contents of the window need to be redrawn.
+  void draw() override;
+  void update() override;
+  void setup() override;
+
+ private:
+  // This will maintain a list of points which we will draw line segments between
+  float mBarrierWidth = 5;
+  vec2 mBallPos;
+  vec2 mBallVel;
+  float mBatLength = 200;
+  float mBallRadius = 13;
+  float height = 600;
+  float width = 800;
+
+  void drawSurroundings();
+  void drawBall();
+  void drawLeftBat();
+  void drawRightBat();
+  void moveBall();
+
+  Rectf mLeftBarrier;
+  Rectf mRightBarrier;
+  Rectf mTopBarrier;
+  Rectf mBottomBarrier;
+
+  Rectf mLeftBat, mRightBat;
+
+  bool mGameOver = false;
+
+  bool ballCollidesDown_p() { return ballCollidesWithRect(mBottomBarrier); }
+  bool ballCollidesRight_p() { return ballCollidesWithRect(mRightBarrier); }
+  bool ballCollidesUp_p() { return ballCollidesWithRect(mTopBarrier); }
+  bool ballCollidesLeft_p() { return ballCollidesWithRect(mLeftBarrier); }
+
+  bool ballCollidesWithRect(Rectf& rect);
+  vec2 advanceBall();
+  void moveLeftBat(bool up_p);
+  void moveRightBat(bool up_p);
+};
+
+void prepareSettings(BasicApp::Settings* settings)
+{
+  settings->setMultiTouchEnabled(false);
+  settings->setWindowSize(800, 600);
+}
+
+void BasicApp::update()
+{
+  static const double timestep = 1.0 / 60.0;
+
+  // Keep track of time.
+  static double time = getElapsedSeconds();
+  static double accumulator = 0.0;
+
+  // Calculate elapsed time since last frame.
+  double elapsed = getElapsedSeconds() - time;
+  time += elapsed;
+  accumulator += math<double>::min(elapsed, 0.1);  // prevents 'spiral of death'
+  while ((accumulator >= timestep) and not mGameOver)
+    {
+      accumulator -= timestep;
+      moveBall();
+      if (mLeftBat.getCenter().y + 200 < mBallPos.y)
```

Comment:
Please dont use directly magic numbers (`200`). I don't know what it refers to, nor the unit. I also don't know (I can guess) if this 200 should always be equal to the 200 a few lines below: if one changes,  should the other one also be updated? Instead of using 200 directly, assign it to a static const variable with a meaning name